shared/common-adapters/confirm-modal/index.js

```diff
@@ -0,0 +1,126 @@
+// @flow
+import * as React from 'react'
+import {Box, Box2, ButtonBar, HeaderOrPopup, Icon, Text, WaitingButton} from '..'
+import * as Styles from '../../styles'
+import type {IconType} from '../icon.constants'
+
+// generally one of icon or header will be given
+export type Props = {|
+  confirmText?: string,
+  content?: React.Node,
+  description: string,
+  header?: React.Node,
+  icon?: IconType,
+  onCancel: () => void,
+  onConfirm: () => void,
+  prompt: string,
+  waitingKey?: string,
+|}
+
+class _ConfirmModal extends React.Component<Props> {
+  render() {
+    return (
+      <Box style={styles.mobileFlex}>
+        <Box2 direction="vertical" style={styles.container}>
+          <Box2
+            alignItems="center"
+            direction="vertical"
+            fullWidth={true}
+            fullHeight={true}
+            style={styles.inner}
+          >
+            {this.props.icon && (
+              <Icon
+                boxStyle={styles.icon}
+                color={Styles.globalColors.black_50}
+                fontSize={Styles.isMobile ? 64 : 48}
+                style={styles.icon}
+                type={this.props.icon}
+              />
+            )}
+            {this.props.header && (
+              <Box2 alignItems="center" direction="vertical" style={styles.icon}>
+                {this.props.header}
+              </Box2>
+            )}
+            <Text center={true} style={styles.text} type="HeaderBig">
+              {this.props.prompt}
+            </Text>
+            <Text center={true} style={styles.text} type="Body">
+              {this.props.description}
+            </Text>
+            {this.props.content}
+          </Box2>
+        </Box2>
+        <Box2 direction="horizontal" style={styles.buttonBox}>
+          <ButtonBar direction="row" fullWidth={true} style={styles.buttonBar}>
+            {!Styles.isMobile && (
+              <WaitingButton
+                fullWidth={true}
+                type="Secondary"
+                label="Cancel"
+                onClick={this.props.onCancel}
+                waitingKey={this.props.waitingKey}
+              />
+            )}
+            <WaitingButton
+              fullWidth={true}
```

Comment:
This and above: `fullWidth` here means the full width available in its containing Box. Since there are two, I think this should just be `flex: 1`. Otherwise when the content changes, the buttons' width change too. (You can tell by switching to different confirm-modal stories.)

Reply:
I was wondering about that, thanks, should be ok now

Reply:
:white_check_mark: